src/Assert.php

```diff
@@ -817,6 +818,24 @@ public static function uuid($value, $message = '')
         }
     }
 
+    public static function throws(callable $expression, $expected, $message = '')
+    {
+        static::string($expected);
+
+        try {
+            $expression();
+        } catch (Exception $e) {
+            if (is_a($e, $expected, true)) {
```

Comment:
interesting! Now I'm wondering what is better, because exceptions are (very) often catched based on their inheritance tree. 
You might want to verify that "any exceptions inheriting \Some\Lib\BaseException" is thrown.
If you want to check a specific exception, you'd then explicitly use a leaf of the inheritance tree (no guarantee of it beeing a subclass indeed).
Otherwise we could propose `Assert::throws` and `Assert::throwsExaclty` (or something like that) but it starts being complicated.

I'd personally be in favor of keeping it like that, since it looks a more instinctive behavior to me, but I'm open to change :)